### PR TITLE
Add library `xod-dev/ds-rtc`

### PR DESCRIPTION
**Pull request by gabbapeople**
The rework of XOD `ds1307` common-hardware nodes in order to make a `datetime` compatible RTC library.

### CODE REVIEW

File: workspace/__lib__/xod-dev/ds-rtc/input-rtc-device/patch.xodp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/input-rtc-device/patch.xodp]: This one is auto-generated and should not be there. @evgenykochetkov @brusherru Do you have an idea how the file was even created on saving?

File: workspace/__lib__/xod-dev/ds-rtc/output-rtc-device/patch.xodp [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/output-rtc-device/patch.xodp]: Ditto

File: workspace/__lib__/xod-dev/ds-rtc/project.xod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "Support for the DS1302/DS1307/DS3231 based breakout RTC boards"
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/project.xod]: It’s not a name, but the description. Version, license is missing. Use other’s library `project.xod` as a template.

File: workspace/__lib__/xod-dev/ds-rtc/rtc-device/patch.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    auto addr = getValue<input_ADDR>(ctx);
+    auto Wire = getValue<input_I2C>(ctx);
+
+    Type dev;
+    dev.addr = addr;
+    dev.wire = Wire;
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/rtc-device/patch.cpp]: Just a bore. There’s a shorter and arguably more clear version:

```cpp
Type dev;
dev.addr = getValue<input_ADDR>(ctx);
dev.wire = getValue<input_I2C>(ctx);
```

File: workspace/__lib__/xod-dev/ds-rtc/rtc-device/patch.xodp [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+  "description": "Constructs a rtc-device value for a DS1302/DS1307/DS3231 RTC board",
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/rtc-device/patch.xodp]: :scroll: an \`rtc-device\` value

File: workspace/__lib__/xod-dev/ds-rtc/unpack-rtc-device/patch.cpp [hunk 6]
```diff
@@ -0,0 +1,13 @@
+
+struct State {
+};
+
+{
+    { GENERATED_CODE }
+}
```
Comment (nkrkv) [inline workspace/__lib__/xod-dev/ds-rtc/unpack-rtc-device/patch.cpp]: Damaged formatting

File: workspace/__lib__/xod-dev/ds-rtc/input-rtc-device/patch.xodp (hunk 1 above)
Comment (gabbapeople) [reply workspace/__lib__/xod-dev/ds-rtc/input-rtc-device/patch.xodp]: That's my mistake.

**Comment by gabbapeople** [comment]
Requested changes have been made
@nkrkv please review

**Comment by gabbapeople** [comment]
@nkrkv Done. Marked as deprecated.

**Pull request merged**